### PR TITLE
Exclude option select filters from facet list

### PR DESCRIPTION
**Pull request by andysellick**
We recently added a new form field to the `option select` component...

Screen Shot 2019-02-27 at 10 11 08

Unfortunately the way the finders JS works is by serialising the whole form, which unintentionally included these inputs, resulting in `&filter=&filter=&filter=` being added to the URL and causing further problems.

This PR firstly modifies the `option select` component to create the filter input using JS (to solve this problem when JS is disabled, previously it was only hiding the element), and secondly updates the `live_search.js` code to explicitly filter out the `option select` filter inputs.

Huge thanks to @aliuk2012 for pairing on this 👍

### CODE REVIEW

File: app/views/finders/_select_facet.html.erb [hunk 1]
```diff
@@ -6,7 +6,8 @@
     :aria_controls_id => "js-search-results-info",
     :options_container_id => select_facet.key,
     :options => select_facet.options("js-search-results-info", select_facet.key),
-    :closed_on_load => close_facet
+    :closed_on_load => close_facet,
+    :show_filter => true
```
Comment (aliuk2012) [inline app/views/finders/_select_facet.html.erb]: I know why you added this but we should remove it.

File: app/assets/javascripts/components/option-select.js [hunk 2]
```diff
@@ -43,7 +43,12 @@
       }
     }
 
-    if (this.$filter.length) {
+    if (this.hasFilter.length) {
+      var filterEl = document.createElement('div');
+      filterEl.innerHTML = this.hasFilter;
+      this.$optionList.prepend($('<div/>').addClass('app-c-option-select__filter').html($(filterEl.childNodes[0].nodeValue)));
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/components/option-select.js]: Will reword, thanks.

File: app/assets/javascripts/live_search.js [hunk 3]
```diff
@@ -63,16 +63,21 @@
...
-  };
+  }
 
   LiveSearch.prototype.getTaxonomyFacet = function getTaxonomyFacet() {
     this.taxonomy = this.taxonomy || new GOVUK.TaxonomySelect({ $el: $('.app-taxonomy-select') });
     return this.taxonomy;
-  }
+  };
+
+  LiveSearch.prototype.getSerializeForm = function getSerializeForm(){
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/live_search.js]: We already have tests that check the creation and modifying of the state, which rely on this function, so I think we're okay?

File: app/views/finders/_select_facet.html.erb (hunk 1 above)
Comment (andysellick) [reply app/views/finders/_select_facet.html.erb]: Thanks, good spot. I added it so I could test all of this in the browser in a finder, not just in the component guide.

**Comment by aliuk2012** [comment]
We should get someone else to review this too

**Pull request merged**